docs/docs/getting-started/overview/package-library.md

```diff
@@ -0,0 +1,121 @@
+
+#### `dot/core`
+
+- The **core package** implements the [Core Service](/getting-started/overview/host-architecture#core-service) -  responsible for block production and block finalization (consensus) and processing messages received from the [Network Service](/getting-started/overview/host-architecture/#network-service).
+
+#### `dot/network`
+
+- The **network package** implements the [Network Service](/getting-started/overview/host-architecture/#network-service) - responsible for coordinating network host and peer interactions. It manages peer connections, receives and parses messages from connected peers and handles each message based on its type.
+
+#### `dot/state`
+
+- The **state package** implements the [State Service](/getting-started/overview/host-architecture#state-service) - the source of truth for all chain and node state that is made accessible to all node services.
+
+#### `dot/sync`
+
+- The **sync package** implements handling of blocks received from the network layer.
+
+#### `dot/rpc`
+
+- The **rpc package** implements the [RPC Service](/getting-started/overview/host-architecture#rpc-service) - an HTTP server that handles state interactions.
+
+#### `dot/types`
+
+- The **types package** implements types defined within the Polkadot Host specification.
+
+## lib packages
+
+#### `lib/babe`
+
+- the **babe package** implements the BABE block production algorithm.
+
+#### `lib/blocktree`
+
+- the **blocktree package** implements the blocktree, a data structure which tracks the chain and all its non-finalized forks.
+
+#### `lib/common`
+
+- the **common package** defines common types and functions.
+
+#### `lib/crypto`
+
+- the **crypto package** contains the key types used by the node (sr25519, ed25519, secp256k1).
+
+#### `lib/grandpa`
+
+- the **grandpa package** implements the GRANDPA finality gadget.
+
+#### `lib/keystore`
+
+- the **keystore package** manages the keystore and includes test keyrings.
+
+#### `lib/runtime`
+
+- the **runtime package** contains various wasm interpreters used to interpret the runtime. It currently contains `life`, `wasmer`, and `wasmtime`; however, `wasmer` is the only interpreter that is fully supported at the moment. In the future, all interpreters will be fully supported.
+
+#### `lib/scale`
+
+- the **scale package** implements the SCALE codec.
+
+#### `lib/services`
+
+- the **services package** implements a common interface for node services.
+
+#### `lib/transaction`
+
+- the **transaction package** is contains transaction types and the transaction queue data structure.
+
+#### `lib/trie`
+
+- the **trie package** implements a modified merkle-patricia trie.
+
+#### `lib/utils`
+
+- the **utils package** is used to manage node and test directories.
+
+## test packages
+
+#### `?`
```

Comment:
Please update accordingly

Reply:
@noot What content do we put here?

Reply:
we can add the polkadot.js, rpc, and stress integration tests, just a quick overview is fine

Reply:
@RyRy79261 you need to describe the existing test package the can be found [here](https://github.com/ChainSafe/gossamer/tree/development/tests). The same way the above packages are described.

Reply:
@dutterbutter just updated as described